### ISSUE TITLE
About the number of training steps

### ISSUE THREAD

**Issue by o0t1ng0o** [fuying-wang/MGCA#26]
Hi @fuying-wang,

I have som questions about counting the training_steps and effective batch size in 'num_training_steps' in mgca_module.py.
Do you count the training steps for each gpu? And args.batch_size indicates the per gpu batch size?
What is the meaning of effective batch size?
 Thank you very much!
https://github.com/HKU-MedAI/MGCA/blob/b9ec84f378e9561628c30f39d5ee64250f450f37/mgca/models/mgca/mgca_module.py#L451

**Comment by fuying-wang** [comment]
Hello,

Training steps denote the total number of steps of updating the model parameters. It is calcualted for all GPUs.
Yes, args.batch_size is the batch size per GPU. Effective batch size is the equivalent batch size of updating the model when taking accumulating gradients and DDP into consideration.

**Comment by o0t1ng0o** [comment]
Thank you for your prompt and detailed reply.
I understand it from your explaination.
